aten/src/ATen/native/DispatchStub.cpp

```diff
@@ -60,6 +69,29 @@ static CPUCapability compute_cpu_capability() {
 #endif
   }
 #endif
+#if defined(__linux__) && defined(HAVE_SVE_CPU_DEFINITION)
+  if (cpuinfo_initialize()) {
+      int ret = prctl(PR_SVE_GET_VL);
```

Comment:
Is this is a syscall? If so, it should be cached. Also, aren't https://github.com/pytorch/cpuinfo provides this information already?

Reply:
prctl is a syscall which we are using to get vector length of SVE from the hardware. Currently this functionality isn't available in cpuinfo library. We have implemented caching for this syscall as requested so that it is only called once and also handled different scenarios (Like what if it fails).
**Refer:** https://github.com/pytorch/pytorch/pull/119571/commits/ceecb567994b8238501fc1386af8ab78235a05a2

On the other suggestion from our conversations in slack to implement the functionality of getting VL from cpuinfo ourselves, can we do this change and track seperately independent of this PR if the above changes makes sense for now.

Reply:
@malfet Thanks for the review. Are there any other comments for us to incorporate?